lib/ansible/modules/network/cnos/cnos_vrf.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+#
+# Copyright (C) 2019 Lenovo.
+# (c) 2017, Ansible by Red Hat, inc
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Module to work on management of local users on Lenovo CNOS Switches
+# Lenovo Networking
+#
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: cnos_vrf
+version_added: "2.8"
+author: "Anil Kumar Muraleedharan (@amuraleedhar)"
+short_description: Manage VRFs on Lenovo CNOS network devices
+description:
+  - This module provides declarative management of VRFs
+    on Lenovo CNOS network devices.
+notes:
+  - Tested against CNOS 10.9.1
+options:
+  name:
+    description:
+      - Name of the VRF.
+    required: true
+  rd:
+    description:
+      - Route distinguisher of the VRF
+  interfaces:
+    description:
+      - Identifies the set of interfaces that
+        should be configured in the VRF. Interfaces must be routed
+        interfaces in order to be placed into a VRF. The name of interface
+        should be in expanded format and not abbreviated.
+  associated_interfaces:
+    description:
+      - This is a intent option and checks the operational state of the for
+        given vrf C(name) for associated interfaces. If the value in the
+        C(associated_interfaces) does not match with the operational state of
+        vrf interfaces on device it will result in failure.
+    version_added: "2.5"
```

Comment:
@amuraleedhar can you remove the `version_added` field as module field is added in `2.8`

Reply:
Removed.